src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,167 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int EMPTY_LIST = 0;
+    private Node head;
+    private Node tail;
+    private int size;
+
+    public MyLinkedList() {
+        this.head = null;
+        this.tail = null;
+        this.size = EMPTY_LIST;
+    }
+
     @Override
     public void add(T value) {
+        Node newNode = new Node(value, null, null);
+        if (size == EMPTY_LIST) {
+            head = newNode;
+        } else {
+            tail.next = newNode;
+            newNode.prev = tail;
+        }
+        tail = newNode;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index > size || index < EMPTY_LIST) {
+            throw new IndexOutOfBoundsException("Index is out of list");
+        }
+
+        if (index == size) {
+            add(value);
+        } else {
+            Node current = findNodeByIndex(index);
+            Node newNode = new Node(value, current.prev, current);
+            if (current.prev != null) {
+                current.prev.next = newNode;
+            }
+            current.prev = newNode;
+            if (index == 0) {
+                head = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list == null || list.isEmpty()) {
+            return;
+        }
+
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
+        if (index >= size || index < EMPTY_LIST) {
```

Comment:
This check is repeated 3 times so it is worth to extract it to a separate method.